.github/workflows/ci.yml

```diff
@@ -7,14 +7,69 @@ on:
   pull_request:
     branches:
       - master
+  workflow_dispatch:
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
+  # different events on the same ref can run in parallel
+  # different refs on the same event can run in parallel
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name}}
   cancel-in-progress: true
 
+env:
+  docs_artifact: docs
+
 jobs:
+  context:
+    runs-on: ubuntu-latest
+
+    outputs:
+      # All github action outputs are strings, even if set to "true"
+      # so when using these values always assert against strings or convert from json
+      # \$\{{ needs.context.outputs.is_fork == 'true' }} // true
+      # \$\{{ fromJson(needs.context.outputs.is_fork) == false }} // true
+      # \$\{{ needs.context.outputs.is_fork == true }} // false
+      # \$\{{ needs.context.outputs.is_fork }} // false
+      is_fork: ${{ steps.context.outputs.is_fork }}
+      is_dependabot: ${{ steps.context.outputs.is_dependabot }}
+      is_push_master: ${{ steps.context.outputs.is_push_master }}
+
+    steps:
+      - name: Log context
+        shell: bash
+        run: |
+          cat <<EOF
+          ${{ toJSON(github) }}
+          EOF
+      - name: Set context
+        id: context
+        shell: bash
+        run: |
+          is_push_master="${{ github.event_name == 'push' && github.ref_name == 'master' }}"
+          is_dependabot="${{ startsWith(github.head_ref, 'dependabot')}}"
```

Comment:
Should we also check that we're not on a fork here ? It doesn't matter now, but it would avoid us ever accidentally considering a branch on someone's fork as a `dependabot` PR, in case we add more logic around `is_dependabot` later.

(My own workaround that only checked the `actor` name suffered from this issue and I didn't like that)

Reply:
I mean specifically to figure out the `is_dependabot` variable value. It should never be true in forks IMHO, regardless of how they have named their branches.

Reply:
I disagree, we have 2 variables 1 tells you if it is a fork, one if it is a dependabot PR. If you want to know if it is a fork/dependabot, just combine the conditions, just like we do in the build.

I think adding an implicit filter for forks on is_dependabot makes the logic less clear.

Reply:
Rebased this PR on a dependabot PR, let's see if the user gives us an answer https://github.com/mozilla/addons-server/pull/22351

Reply:
Pushed a commit using the `github.event.pull_request.user.login` to match the dependabot user.. this should work.

Reply:
I actually think your original way is the best way, it is in the official documentation. Combining that with the new logic for forks, we should be good to go. https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions

Wdyt?

Reply:
But as it stands the condition is not specific enough, it does not guarantee that it's a dependabot PR: Any branch (from a fork or not) can start with `dependabot`, we could accidentally call a branch `dependabot-fix` or `dependabot-config` ourselves...

At the very least we should make that condition true if the branch name starts with `dependabot/` (with the slash) to avoid this IMHO.

Reply:
The problem with my original condition is that it prevents the `Build Docker image` job from running on commits from dependabot in `master`. Using the branch named worked around that.

I'm fine with going back to checking the `actor` but we need another check so that `Build Docker image` runs on `master` from the main repos no matter what

Reply:
Yes, the check in the build step needs to be made more specific. I also found while testing this it is a bit confusing if the build doesn't run, that you don't know exactly why. So I've added a step that expalins the workflows reasoning as it goes. Wdyt?

Reply:
Found a better way to check for default branch and split the branch we are on from the event type as they can be considered independently in many cases. I will retest everything as it is siginficantly changed.